vnext/ReactUWP/Views/SIPEventHandler.cpp

```diff
@@ -20,38 +21,85 @@ namespace react {
+// TryShow and TryHide works on >= RS5
+
+void SIPEventHandler::AttachView(XamlView xamlView, bool fireKeyboardEvents) {
+  m_fireKeyboradEvents = fireKeyboardEvents;
+
+  if (!IsRS3OrHigher()) {
+    return; // CoreInputView is only supported on >= RS3.
+  }
+
+  if (Is19H1OrHigher()) {
+    // 19H1 and higher supports island scenarios
+    auto uiElement(xamlView.as<winrt::UIElement>());
+    m_coreInputView = winrt::CoreInputView::GetForUIContext(uiElement.UIContext());
+  } else {
+    m_coreInputView = winrt::CoreInputView::GetForCurrentView();
+  }
+
+  if (m_coreInputView) {
+    auto occlusions = m_coreInputView.GetCoreInputViewOcclusions();
+    m_isShowing = !IsOcclusionsEmpty(occlusions);
+    m_occlusionsChanged_revoker = m_coreInputView.OcclusionsChanged(
+        winrt::auto_revoke, [this](auto &&, const winrt::CoreInputViewOcclusionsChangedEventArgs &e) {
           if (!e.Handled()) {
-            winrt::Rect finalRect = winrt::RectHelper::Empty();
-            winrt::IVectorView<winrt::CoreInputViewOcclusion> occlusions = e.Occlusions();
-            for (uint32_t i = 0; i < occlusions.Size(); i++) {
-              winrt::CoreInputViewOcclusion occlusion = occlusions.GetAt(i);
-              if (occlusion.OcclusionKind() == winrt::CoreInputViewOcclusionKind::Docked) {
-                finalRect = winrt::RectHelper::Union(finalRect, occlusion.OccludingRect());
+            bool wasShowing = m_isShowing;
+            m_isShowing = !IsOcclusionsEmpty(e.Occlusions());
+            if (wasShowing != m_isShowing && m_fireKeyboradEvents) {
+              if (!m_isShowing) {
+                folly::dynamic params = folly::dynamic::object("screenY", 0)("screenX", 0)("width", 0)("height", 0);
+                SendEvent("keyboardDidHide", std::move(params));
+              } else {
+                folly::dynamic params = folly::dynamic::object(
+                    "endCoordinates",
+                    folly::dynamic::object("screenY", m_finalRect.Y)("screenX", m_finalRect.X)(
+                        "width", m_finalRect.Width)("height", m_finalRect.Height));
+                SendEvent("keyboardDidShow", std::move(params));
               }
             }
-
-            if (winrt::RectHelper::GetIsEmpty(finalRect)) {
-              folly::dynamic params = folly::dynamic::object("screenY", 0)("screenX", 0)("width", 0)("height", 0);
-              SendEvent("keyboardDidHide", std::move(params));
-            } else {
-              folly::dynamic params = folly::dynamic::object(
-                  "endCoordinates",
-                  folly::dynamic::object("screenY", finalRect.Y)("screenX", finalRect.X)("width", finalRect.Width)(
-                      "height", finalRect.Height));
-              SendEvent("keyboardDidShow", std::move(params));
-            }
           }
         });
   }
 }
 
-SIPEventHandler::~SIPEventHandler() {
-  m_occlusionsChanged_revoker = {};
+bool SIPEventHandler::TryShow() {
```

Comment:
>TryShow [](start = 22, length = 7)

I didn't see you can TryShow, why do you need this function?

Reply:
I am expecting to call this function in one of the future ScrollView's API implementation. For now, I will just comment it out.

---
In reply to: [348697998](https://github.com/microsoft/react-native-windows/pull/3665#discussion_r348697998) [](ancestors = 348697998)